### PR TITLE
fix: Overloading of xsmm invoke calls doesn't work

### PR DESCRIPTION
**Pull request by KavithaTipturMadhu**

### CODE REVIEW

File: include/Standalone/Dialect/Xsmm/XsmmOps.td [hunk 1]
```diff
@@ -46,6 +46,15 @@ def Xsmm_TernaryOp : Xsmm_Op<"ternary", []> {
     }
     operand_iterator arg_operand_begin() { return operand_begin(); }
     operand_iterator arg_operand_end() { return operand_end(); }
+
+    std::string getOperandType(){
+    	auto operand = std::next(arg_operand_begin());
+        Type type = (*operand).getType().isa<MemRefType>()?(*operand).getType().cast<MemRefType>().getElementType(): (*operand).getType();
+        if(type.isBF16())
+		return "bf16";
+	assert(type.isF32());
```
Comment (chelini) [inline include/Standalone/Dialect/Xsmm/XsmmOps.td:55]: Please add a message on the assert: `assert(types.isF32() && "expect bf16 or f32")`

File: lib/Standalone/ConvertXsmmToFunc.cpp [hunk 2]
```diff
@@ -19,6 +19,7 @@ using namespace mlir::xsmm;
 
 #define GEN_PASS_CLASSES
 #include "Standalone/Passes.h.inc"
+#include <iostream>
```
Comment (chelini) [inline lib/Standalone/ConvertXsmmToFunc.cpp:22]: Drop this.

File: lib/Standalone/ConvertXsmmToFunc.cpp [hunk 3]
```diff
@@ -96,8 +96,15 @@ struct ConvertTernaryXsmmOp : public OpRewritePattern<TernaryOp> {
 
   LogicalResult matchAndRewrite(TernaryOp ternaryOp,
                                 PatternRewriter &rewriter) const override {
+    std::string funcName =
+        "xsmm_" + stringifyEnum(ternaryOp.getCallee()).str() + "invoke";
+    if (ternaryOp.getOperandType() == "bf16")
+      funcName = funcName + "_bf16";
+    else
+      assert(ternaryOp.getOperandType() == "f32";
```
Comment (chelini) [inline lib/Standalone/ConvertXsmmToFunc.cpp:104]: missing a parenthesis and a message.

Comment (chelini) [reply lib/Standalone/ConvertXsmmToFunc.cpp:104]: Why for bf16 we add a `_bf16` while for f32 only `f32` should we add `_f32`? (to be consistent)

**Review by chelini — APPROVED** [review]
Pleas run `git-clang-format HEAD~1`

File: lib/Standalone/ConvertXsmmToFunc.cpp (hunk 3 above)
Comment (KavithaTipturMadhu) [reply lib/Standalone/ConvertXsmmToFunc.cpp:104]: do you mean the xsmm call should have an _f32 suffix? The above is just a type check, and there are no underscores. Only when composing the function call name, we are appending _bf16.

**Comment by KavithaTipturMadhu** [comment]
> Pleas run `git-clang-format HEAD~1`

Is there some way to run git clang format on .td files? There don't seem to be any changes after running this.

**Comment by chelini** [comment]
No there is not way (unfortunately) to format this. We need to do manually.

File: lib/Standalone/ConvertXsmmToFunc.cpp (hunk 3 above)
Comment (chelini) [reply lib/Standalone/ConvertXsmmToFunc.cpp:104]: Yeah, I would make a _f32 suffix too. Just to be consistent.

**Review by chelini — APPROVED** [review]
Thanks. Much better! Please run clang-format and fix remaining comments. Feel free to merge after that! Cool.

**Pull request merged**